contracts/MockContract.sol

```diff
@@ -363,11 +368,14 @@ contract MockContract is MockInterface {
 			result = fallbackExpectation;
 		}
 
-		// Record invocation
-		invocations += 1;
-		methodIdInvocations[keccak256(abi.encodePacked(resetCount, methodId))] += 1;
-		calldataInvocations[keccak256(abi.encodePacked(resetCount, msg.data))] += 1;
-
+		// Record invocation only if we are not invoked via STATICCALL
+		(bool success, ) = address(this).call(abi.encodeWithSignature("tryWrite()"));
+		if (success) {
+			invocations += 1;
```

Comment:
You could have also put all of this into a method (e.g. `updateInvocationCount(bytes4 methodId, bytes data)`) that has a `require(msg.sender == this, "Can only be called from the contract itself")`. 

Then you could just call `address(this).call(abi.encodeWithSignature("updateInvocationCount(bytes4,bytes)", methodId, msg.data));`

not sure if that would look nicer/ less hacky :)

Reply:
The reason I didn't do that is because I ended up with this warning: `Warning: Return value of low-level calls not used.`

I could silence it by doing something like
```
(, bytes memory r) = address(this).call(...);
 assert(r.length == 0);
```
but this also seemed hacky (and less explicit than calling a method tryWrite). Is there a better way to get rid of this compiler warning?

Reply:
@s1na, I'm sorry about that, I should have communicated clearer. I really appreciate your collaboration. Happy to merge your PR if you prefer.

Did you find a way to not get the compiler warning?

Reply:
I would have just ignored the warning :P (currently there is no way to suppress the warning)

Reply:
@fleupold oh no worries, I just meant it was a funny coincidence that we were working on almost exactly the same solution at the same time ;)

No, I noticed the warning after I saw your comment. Doesn't seem to be harmful.

Reply:
I'd prefer not to pollute developers test outputs with warning in the mock contract if we can avoid it.